subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c

```diff
@@ -1692,11 +1693,92 @@ gst_msdkenc_get_surface_from_pool (GstMsdkEnc * thiz, GstBufferPool * pool,
   return msdk_surface;
 }
 
+static GstMsdkSurface *
```

Comment:
How about changing the name to gst_msdkenc_get_surface_from_pool (the old xxx_from_pool can be removed actually because has_vpp part in the encoder will be removed as well). This func is called when we need to get a buffer from the pool and then do the copy. Also, the copy does not always happen with system memory, other video memory that we don't recognize(like gl memory) can trigger this.

Reply:
Modified, thx.